packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1114,7 +1130,7 @@ export class SnapController extends BaseController<
         }
       }
     }
-    return allEndowments.length > 0 ? allEndowments : undefined;
+    return [...DEFAULT_EXPOSED_APIS, ...allEndowments];
```

Comment:
We should probably dedupe this array as a defensive measure, and log an error to the console if duplicates are found.

Reply:
Hmm, added deduping, why would this need to be logged? 🤔